CMakeLists.txt

```diff
@@ -66,39 +66,44 @@ set(BUILD_SHARED_LIBS ON CACHE BOOL "Build shared")
 # Include cmake scripts
 include(cmake/Dependencies.cmake)
 
-# Detect compiler support for target ID
-# This section is deprecated. Please use rocm_check_target_ids for future use.
-if( CMAKE_CXX_COMPILER MATCHES ".*/hipcc$" )
-    execute_process(COMMAND ${CMAKE_CXX_COMPILER} "--help"
-        OUTPUT_VARIABLE CXX_OUTPUT
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-        ERROR_STRIP_TRAILING_WHITESPACE)
-    string(REGEX MATCH ".mcode\-object\-version" TARGET_ID_SUPPORT ${CXX_OUTPUT})
-endif()
+option(ROCRAND_CMAKE_HIP_LANGUAGE "Use CMake's HIP language support" OFF)
+if(ROCRAND_CMAKE_HIP_LANGUAGE)
+  enable_language(HIP)
```

Comment:
You still need to call `find_package(hip)` when using hip language. Libraries like rocRAND need to provide `hip::host` in the targets interface because it includes hip's runtime API headers.

Reply:
Thanks. I was just about to request your review. I haven't yet tested this with downstream libraries.

Reply:
Ah wait it looks like rocRAND actually needs hip::device as it includes hip's device headers:

https://github.com/ROCmSoftwarePlatform/rocRAND/blob/develop/library/include/rocrand/rocrand.h#L29

We would need two different targets to support using rocRAND with HIP language and with CXX language.

Reply:
It's not actually clear to me why it includes `<hip/hip_device.h>`. rocRAND appears to provide a C99 API. Sadly, [the documentation](https://rocrand.readthedocs.io/) is... lacking.

Reply:
> We would need two different targets to support using rocRAND with HIP language and with CXX language.

Maybe we could detect the language with a generator expression? I'm not entirely sure it's possible, but I was thinking something like:
```cmake
set_target_properties(roc::rocrand
  PROPERTIES
    INTERFACE_LINK_LIBRARIES "$<$<NOT:$<COMPILE_LANGUAGE:HIP>>:hip::device>"
)
```